fdbserver/DataDistribution.actor.cpp

```diff
@@ -719,6 +682,23 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 		teamBuilder.cancel();
 	}
 
+	void addLaggingStorageServer(Key zoneId) {
+		lagging_zones[zoneId]++;
+		if (lagging_zones.size() > std::max(1, configuration.storageTeamSize - 1) && !disableFailingLaggingServers.get())
```

Comment:
I don't quite get why we want to limit the number of lagging SS "per zone"? 

Let's assume zone is a host. 
If the host is a limping machine (that runs slower for some reason), all SSes on the host will have big lags. Why cannot we mark them as unhealthy at the same time so that we can get off the host faster?

I thought we should limit the total number of lagging SSes that are marked as unhealthy, instead of per-zone. 
Did I miss something?

Reply:
Is the point of this to limit the number lagging per zone, or to limit the number of lagging zones? The if looks to be checking the latter, which makes sense if you don't want to have all replicas in a team lagging, etc.

Reply:
I agree that the latter makes sense. 
In that case, what do we want to achieve by keeping the number of lagging servers per zone? Can we use a `std::set` to memorize which zones have lagging SSes?

Reply:
We want to have a backstop mechanism to stop marking a lot of lagging storage servers as undesired which can happen in case of remote data center down for some time and coming back up. As AJ said, the idea is to prevent triggering data movement when all the members in a team are lagging.

Reply:
I keep track of the number of lagging servers per zone to easily determine if a zone got healthy.

Reply:
I see the logic here now. Thanks for explanation!